message Vision_DetectionBall {
  required float  speed_mod        = 1;
  optional float  speed_direction  = 2; 
  optional uint32 area       =  3;
  required float  x          =  4;
  required float  y          =  5;
  optional float  height     =  6;
  optional uint32 ball_state =  7;
  optional uint32 last_touch =  8;
  optional float  next_x     =  9;
  optional float  next_y     =  10;
}

message Vision_DetectionRobot {
  required float  confidence  =  1;
  optional uint32 robot_id    =  2;
  required float  x           =  3;
  required float  y           =  4;
  optional float  orientation =  5;
  optional float  speed_mod         =  6;
  optional float  speed_direction   =  7;
  optional float  next_x      =  8;
  optional float  next_y      =  9;
}

message Vision_DetectionFrame {
  required Vision_DetectionBall  balls         = 1;
  repeated Vision_DetectionRobot robots_yellow = 2;
  repeated Vision_DetectionRobot robots_blue   = 3;
}